vocabulary-draft/continuous-delivery.md

```diff
@@ -0,0 +1,114 @@
+
+### Component Version Created
+
+Announces that a new version of a Product Component has been created.
+
+Examples of Component Versions include:
+
+- New version of a software module or application.
+- New version of a document or piece of configuration.
+- New revision of a hardware component.
+
+The purpose of this event would be for systems to start creating Product Compositions / Baseline Versions that include this Component Version.
+
+Component Version Created Events MUST include the following attributes:
+
+- **Component Version ID**: Unique identifier of component version.
+- **Component ID**: Identifier of component of which this is a version.
+
+Optional attributes:
+
+- **Version Name**: Machine-friendly and human-friendly version tag/number/etc. for component version, e.g. “1.4.2-alpha2”.
+- **Component Name**: Machine-friendly and human-friendly name of component, e.g. “c37-backbone-router”.
+- **Component Display Name**: UX/human-friendly display name of component, e.g. “C37 Backbone Router”.
+- **Artifacts**: List of artifacts (files, packages, Docker images etc.) part of this component version.
+- **Component Track**: States what track (e.g. platform variant (x86_64, ARM etc.), project (e.g. X-DataCenter, X-Cloud etc.) or other) the component, of which this is a version, belongs to.
+
+### Product Composition Created
+
+Announces that a new composition of source-files, artifacts, component versions and other compositions/baseline versions, has been created.
+
+__Note:__ This event description suffers from a missing piece of terminology to represent the “recipe” by that controls what is included in new compositions. E.g. “The latest stable version (LSV) of this application, the LSV of this library, and the LSV of this configuration file”. For now, this will be referred to as a “Recipe” until a better name is given. (ESCI calls this a Baseline, Eiffel likely has a term for this as well).
+
+Product Composition Created Events MUST include the following attributes:
+
+- **Composition ID**: Unique identifier of composition.
+- **Recipe ID**: Identifier of recipe of which this is a composition.
+- **Content**: List of sources, artifacts, components and other compositions part of this composition
+
+Optional attributes:
+
+- **Version Name**: Machine-friendly and human-friendly version tag/number/etc. for composition, e.g. “1.4.2-alpha2”.
+- **Recipe Name**: Machine-friendly and human-friendly name of recipe, e.g. “gemeni-backbone-gen2”.
+- **Recipe Display Name**: UX/human-friendly display name of recipe, e.g. "Gemini Backbone, Gen2”.
+- **Artifacts**: List of artifacts (files, packages, Docker images etc.) part of this component version.
+- **Component Track**: States what track (e.g. platform variant (x86_64, ARM etc.), project (e.g. X-DataCenter, X-Cloud etc.) or other) the component, of which this is a version, belongs to.
+
+### Confidence Label Verdict
+
```

Comment:
@erkist same here again.. we can create an event for this, but how many tools are going to be able to generate this information? Once again, I do see the point of having this, but should we include it in the first go? 
I am thinking about OSS CI/CD systems around the Cloud Native space, and for example inside the CD.foundation and I would love to see how those projects can generate information about this. I would definitely raise this with the `interoperability SIG` for discussions.

Reply:
@salaboy Again, good point. Perhaps a question this time: If this is not something that most other tools would generate, how does this relate to interoperability? Eiffel and ESCI both have this concept (or at least highly related concepts), but I don't think it would be seen as very important for a Cloud Native space to align on this.

Perhaps this is part of a larger question: Can we create a separate "spec extension" to hold these kinds of concepts that are not part of the basic set, and possibly include them later if adoption increases? Sort of like a Python PEP, which can be widely used and aligned on before it is included in the language itself.

Reply:
I commented on this in [Pull Request #8](https://github.com/cdfoundation/sig-events/pull/8#discussion_r588161991)

For me, it makes sense to break out the confidence or trust in an artifact out of the metadata and put it in an event. In this way, you can build an artifact e.g. a docker image, then run tests on it, run security tests on it, and so forth without having to resubmit an event the artifact is available.

The point I am trying to make is that we should only build a docker image once but our confidence in it can increase or decrease as we run tests on it and use it.

Is this making any sense or am I losing people? Do you still think this is too advanced for the first version?

Reply:
@m-linner-ericsson You are making perfect sense to me, and I agree that confidence should be separate from artifacts (both for the reason you stated above, and for the reason that you can have confidence in things other than artifacts). But then I also come from Ericsson initially, so I am likely biased. Nice to see the eBay usage as well!

I would of course like to have all of these events in the spec early on, as I need all of them (or something else that solves the same needs).

Reply:
@m-linner-ericsson does make a lot of sense to me for sure, and @erkist I totally understand the need for all of them, but I wonder where is the line. 

I think that we all agree here that the confidence level event is separated from the artifact itself.